--- origsrc/filezilla-3.0.11.1/configure.in	2008-07-05 09:09:37.000000000 -0500
+++ src/filezilla-3.0.11.1/configure.in	2008-07-16 21:37:15.765625000 -0500
@@ -239,26 +239,7 @@
   # Libidn checks
   # -------------
 
-  AC_CHECK_HEADER(idna.h,, 
-  [
-    AC_MSG_ERROR([idna.h not found which is part of GNU libidn.])
-  ])
-
-  AC_ARG_WITH(idn-lib,
-  	AC_HELP_STRING([--with-idn-lib=FILE], [Use the given path to the idn library.]),
-  [
-	if test "$withval" != "yes" -a "$withval" != ""; then
-		IDN_LIB=$withval
-	fi
-  ])
-
-  if test "x$IDN_LIB" = "x"; then
-    AC_CHECK_LIB(idn, stringprep_check_version, IDN_LIB="-lidn",
-                 AC_MSG_ERROR([GNU libidn not found. Try using --with-idn-lib=FILE to specify the library path.])
-                 )
-  fi
-
-  AC_SUBST(IDN_LIB)
+  PKG_CHECK_MODULES(IDN, libidn)
 
   # Binreloc
   # --------
--- origsrc/filezilla-3.0.11.1/src/interface/Makefile.am	2008-07-05 09:09:39.000000000 -0500
+++ src/filezilla-3.0.11.1/src/interface/Makefile.am	2008-07-16 21:37:15.781250000 -0500
@@ -180,15 +180,15 @@
 endif
 
 filezilla_CPPFLAGS = $(WX_CPPFLAGS)
-filezilla_CXXFLAGS = $(WX_CXXFLAGS_ONLY)
-filezilla_CFLAGS = $(WX_CFLAGS_ONLY)
+filezilla_CXXFLAGS = $(WX_CXXFLAGS_ONLY) $(IDN_CFLAGS)
+filezilla_CFLAGS = $(WX_CFLAGS_ONLY) $(IDN_CFLAGS)
 
 filezilla_LDFLAGS = ../engine/libengine.a
 filezilla_LDFLAGS += ../tinyxml/libtinyxml.a
 filezilla_LDFLAGS += $(LIBGNUTLS_LIBS)
 filezilla_LDFLAGS += $(WX_LIBS)
 filezilla_LDFLAGS += $(RESOURCEFILE)
-filezilla_LDFLAGS += $(IDN_LIB)
+filezilla_LDFLAGS += $(IDN_LIBS)
 
 if MINGW
 filezilla_LDFLAGS += -lole32 -luuid -lnetapi32
--- origsrc/filezilla-3.0.11.1/src/interface/commandqueue.h	2008-07-05 09:09:38.000000000 -0500
+++ src/filezilla-3.0.11.1/src/interface/commandqueue.h	2008-07-16 21:37:15.796875000 -0500
@@ -5,7 +5,7 @@
 class CNotification;
 class CMainFrame;
 
-DECLARE_EVENT_TYPE(fzEVT_GRANTEXCLUSIVEENGINEACCESS, -1)
+DECLARE_LOCAL_EVENT_TYPE(fzEVT_GRANTEXCLUSIVEENGINEACCESS, -1)
 
 class CCommandQueue
 {
--- origsrc/filezilla-3.0.11.1/src/interface/filelistctrl.cpp	2008-07-05 09:09:39.000000000 -0500
+++ src/filezilla-3.0.11.1/src/interface/filelistctrl.cpp	2008-07-16 21:37:15.812500000 -0500
@@ -10,8 +10,8 @@
 #include "filelist_statusbar.h"
 
 #ifndef __WXMSW__
-DECLARE_EVENT_TYPE(fz_EVT_FILELIST_FOCUSCHANGE, -1)
-DECLARE_EVENT_TYPE(fz_EVT_DEFERRED_MOUSEEVENT, -1)
+DECLARE_LOCAL_EVENT_TYPE(fz_EVT_FILELIST_FOCUSCHANGE, -1)
+DECLARE_LOCAL_EVENT_TYPE(fz_EVT_DEFERRED_MOUSEEVENT, -1)
 #ifndef FILELISTCTRL_INCLUDE_TEMPLATE_DEFINITION
 DEFINE_EVENT_TYPE(fz_EVT_FILELIST_FOCUSCHANGE)
 DEFINE_EVENT_TYPE(fz_EVT_DEFERRED_MOUSEEVENT)
--- origsrc/filezilla-3.0.11.1/src/interface/locale_initializer.cpp	2008-07-05 09:09:38.000000000 -0500
+++ src/filezilla-3.0.11.1/src/interface/locale_initializer.cpp	2008-07-16 21:37:15.828125000 -0500
@@ -174,12 +174,14 @@
 		return file;
 
 
+#ifdef ENABLE_BINRELOC
 	file = CheckPathForDefaults(mkstr(SELFPATH), 2, "share/filezilla/fzdefaults.xml");
 	if (!file.empty())
 		return file;
 	file = CheckPathForDefaults(mkstr(DATADIR), 0, "filezilla/fzdefaults.xml");
 	if (!file.empty())
 		return file;
+#endif
 
 	std::string path = mkstr(getenv("PATH"));
 	while (!path.empty())
--- origsrc/filezilla-3.0.11.1/tests/Makefile.am	2008-07-05 09:09:37.000000000 -0500
+++ src/filezilla-3.0.11.1/tests/Makefile.am	2008-07-16 21:37:15.843750000 -0500
@@ -9,12 +9,12 @@
 test_SOURCES = test.cpp dirparsertest.cpp serverpathtest.cpp
 
 test_CPPFLAGS = $(WX_CPPFLAGS)
-test_CXXFLAGS = $(WX_CXXFLAGS_ONLY) $(CPPUNIT_CFLAGS)
+test_CXXFLAGS = $(WX_CXXFLAGS_ONLY) $(CPPUNIT_CFLAGS) $(IDN_CFLAGS)
 
 test_LDFLAGS = $(CPPUNIT_LIBS)
 test_LDFLAGS += ../src/engine/libengine.a 
 test_LDFLAGS += $(LIBGNUTLS_LIBS)
 test_LDFLAGS += $(WX_LIBS)
-test_LDFLAGS += $(IDN_LIB)
+test_LDFLAGS += $(IDN_LIBS)
 
 test_DEPENDENCIES = ../src/engine/libengine.a
